gdal/ogr/ogrsf_frmts/lvbag/ogrlvbaglayer.cpp

```diff
@@ -0,0 +1,444 @@
+            }
+
+            osElementString += " ";
+            osElementString += papszIter[0];
+            osElementString += "=\"";
+            osElementString += papszIter[1];
+            osElementString += "\"";
+            papszIter += 2;
+        }
+
+        osElementString += ">";
+    }
+    else if( nFeatureCollectionDepth > 0 && nFeatureElementDepth == 0 &&
+             EQUAL("sl-bag-extract:bagObject", pszName) )
+    {
+        nFeatureElementDepth = nCurrentDepth;
+        if( !bSchemaOlny )
+            poFeature = new OGRFeature(poFeatureDefn);
+    }
+    else if( nFeatureCollectionDepth == 0 && EQUAL("sl:standBestand", pszName) )
+    {
+        nFeatureCollectionDepth = nCurrentDepth;
+    }
+
+    nCurrentDepth++;
+}
+
+/************************************************************************/
+/*                           endElementCbk()                            */
+/************************************************************************/
+
+void OGRLVBAGLayer::EndElementCbk(const char *pszName)
+{
+    nCurrentDepth--;
+
+    if( nCurrentDepth > nAttributeElementDepth && nAttributeElementDepth > 0
+    && nGeometryElementDepth == 0 )
+    {
+        const char *pszTag = XMLTagSplit(pszName);
+        
+        StopDataCollect();
+        if ( !osElementString.empty() ) 
+        {
+            if( bSchemaOlny )
+            {
+                OGRFieldType eType = OFTString;
+                OGRFieldDefn oField(pszTag, eType);
+                poFeatureDefn->AddFieldDefn(&oField);
+            }
+            else
+            {
+                const int iFieldIndex = poFeatureDefn->GetFieldIndex(pszTag);
+                if( iFieldIndex > -1 )
+                {
+                    poFeature->SetField(iFieldIndex, osElementString.c_str());
+                }
+            }
+        }
+        osElementString.Clear();
+    }
+    else if( nAttributeElementDepth == nCurrentDepth )
+    {
+        nAttributeElementDepth = 0;
+    }
+    else if( nGeometryElementDepth > 0 && nCurrentDepth > nGeometryElementDepth )
+    {
+        osElementString += "</";
+        osElementString += pszName;
+        osElementString += ">";
+    }
+    else if( nGeometryElementDepth == nCurrentDepth )
+    {
+        StopDataCollect();
+        if( !osElementString.empty() )
+        {
+            std::unique_ptr<OGRGeometry> poGeom = std::unique_ptr<OGRGeometry>{
+                CPL_REINTERPRET_CAST(OGRGeometry *, OGR_G_CreateFromGML(osElementString.c_str())) };
+            if( poGeom && !poGeom->IsEmpty() )
+            {
+                if( !bSchemaOlny )
+                    poFeature->SetGeometryDirectly(poGeom.release());
+                else
+                {
+                    OGRGeomFieldDefn *poGeomField = poFeatureDefn->GetGeomFieldDefn(0);
+                    poGeomField->SetSpatialRef(poGeom->getSpatialReference());
+                    poGeomField->SetType(poGeom->getGeometryType());
```

Comment:
could it happen that there would be heterogenous geometry types in the layer ? in which case you'd want to assign wkbUnknown

Reply:
The geom type can be different per datasource (often polygon/point) but stays consistent per layer.